src/sim_adapter/uit_adapter.py

```diff
@@ -0,0 +1,64 @@
+from .simulator import SimAdapter, Simulator
+from geometry_msgs.msg import Twist
+import numpy as np
+import cv2
+
+import eventlet
+import socketio
+
+from PIL import Image
+from io import BytesIO
+import base64
+
+class UITSimAdapter(SimAdapter):
+    r"""
+    This adapter is aimed to work with UIT-RaceCar's competition simulators.
+
+    These simulators use `socketio` to connect and interact with our platform.
+    """
+
+    def __init__(self, sim: Simulator):
+        self.sim = sim
+
+        self.sio = socketio.Server()
+        self.app = socketio.WSGIApp(self.sio)
+
+        self.speed = 0.0
+        self.angle = 0.0
+
+        self.sio.on('telemetry', self.telemetry)
+        self.sio.on('connect', self.connect)
+
+        eventlet.wsgi.server(eventlet.listen(('', 4567)), self.app)
```

Comment:
Instead of hard-coding a port number, try reading from the parameter server for flexibility

Reply:
And also for the hostname, I think even if the host is empty, which is interpreted as `0.0.0.0`, we should also support a way to change it for future use

Reply:
I've add parameter for hostname and port.

Reply:
Btw I've add parameter to select different adapters.